src/test/scala/firrtlTests/ZeroWidthTests.scala

```diff
@@ -0,0 +1,98 @@
+// See LICENSE for license details.
+
+package firrtlTests
+
+import org.scalatest.Matchers
+import java.io.{StringWriter,Writer}
+import firrtl.ir.Circuit
+import firrtl._
+import firrtl.Parser.IgnoreInfo
+import firrtl.Parser
+import firrtl.passes._
+
+class ZeroWidthTests extends FirrtlFlatSpec {
+  val passes = Seq(
+      ToWorkingIR,
+      ResolveKinds,
+      InferTypes,
+      ResolveGenders,
+      InferWidths,
+      ZeroWidth)
+  private def exec (input: String) = {
+    passes.foldLeft(parse(input)) {
+      (c: Circuit, p: Pass) => p.run(c)
+    }.serialize
+  }
+   // =============================
+  "Zero width port" should " be deleted" in {
+    val input =
+      """circuit Top :
+        |  module Top :
+        |    input y : UInt<0>
+        |    output x : UInt<1>
+        |    x <= y""".stripMargin
+    val check =
+      """circuit Top :
+         |  module Top :
+         |    output x : UInt<1>
+         |    x <= UInt<1>(0)""".stripMargin
+      (parse(exec(input))) should be (parse(check))
+  }
+  "Add of <0> and <2> " should " be <3>" in {
+    val input =
+      """circuit Top :
+        |  module Top :
+        |    input y : UInt<0>
+        |    output x : UInt
+        |    x <= add(y, UInt<2>(2))""".stripMargin
+    val check =
+      """circuit Top :
+        |  module Top :
+        |    output x : UInt<3>
+        |    x <= add(UInt<1>(0), UInt<2>(2))""".stripMargin
+      (parse(exec(input)).serialize) should be (parse(check).serialize)
+  }
+  "Mux on <0>" should "constprop" in {
+    val input =
+      """circuit Top :
+        |  module Top :
+        |    input y : UInt<0>
+        |    output x : UInt
+        |    x <= mux(y, UInt<2>(2), UInt<2>(1))""".stripMargin
+    val check =
+      """circuit Top :
+        |  module Top :
+        |    output x : UInt<2>
+        |    x <= UInt<2>(1)""".stripMargin
+      (parse(exec(input)).serialize) should be (parse(check).serialize)
+  }
+}
```

Comment:
I'd like to see a test where a zero width wire is in a Bundle and one with it in a Vec